File: ruby/codewars/7kyu/complementary_dna.rb
# https://www.codewars.com/kata/554e4a2f232cdd87d9000038/train/ruby

def DNA_strand(dna)
  dna.tr('ATCG', 'TAGC')
end

p DNA_strand("AAAA") =="TTTT"
p DNA_strand("ATTGC") == "TAACG"
p DNA_strand("GTAT") =="CATA"

File: go/codewars/7kyu/complementary_dna.go
package main

import "fmt"

func DNAStrand(dna string) string {
	var result string
	for _, v := range dna {
		switch v {
		case 'A':
			result += "T"
		case 'T':
			result += "A"
		case 'C':
			result += "G"
		case 'G':
			result += "C"
		}
	}
	return result
}

func main() {
	fmt.Println(DNAStrand("AAAA") == "TTTT")
}
